Sources/NIOCore/EventLoop.swift

```diff
@@ -220,7 +263,47 @@ public struct EventLoopIterator: Sequence, IteratorProtocol {
 public protocol EventLoop: EventLoopGroup {
     /// Returns `true` if the current `NIOThread` is the same as the `NIOThread` that is tied to this `EventLoop`. `false` otherwise.
     var inEventLoop: Bool { get }
+    
+    #if swift(>=5.6)
```

Comment:
As a stylistic note, it might be better to put the `#if...#else` around every method. It'll make it easier to keep track of whether we've appropriately duplicated the methods.

Reply:
Done